kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm-queries/src/main/java/org/kie/server/remote/rest/jbpm/queries/ProcessInstanceQueryResource.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.remote.rest.jbpm.queries;
+
+import static org.kie.server.api.rest.RestURI.*;
+import static org.kie.server.remote.rest.common.util.RestUtils.*;
+
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.kie.server.api.model.instance.ProcessInstanceList;
+import org.kie.server.remote.rest.common.Header;
+import org.kie.server.services.api.KieServerRegistry;
+import org.kie.server.services.jbpm.queries.ProcessInstanceQueryServiceBase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("server/" + PROCESS_INSTANCES_GET_FILTERED_URI)
+public class ProcessInstanceQueryResource {
+	
+private static final Logger logger = LoggerFactory.getLogger(TaskQueryResource.class);
+	
+	private final ProcessInstanceQueryServiceBase processInstanceQueryServiceBase;
+	private KieServerRegistry context;
+	
+	public ProcessInstanceQueryResource(ProcessInstanceQueryServiceBase processInstanceQueryServiceBase, KieServerRegistry context) {
+		this.processInstanceQueryServiceBase = processInstanceQueryServiceBase;
+		this.context = context;
+	}
+	
+	
+	//TODO: We now only support standard QueryFilterSpec configurations ... Do we also need to support builders???
+	@POST
+	@Produces({MediaType.APPLICATION_XML, MediaType.APPLICATION_JSON})
+	public Response getProcessInstancesWithFilters(@Context HttpHeaders headers, 
```

Comment:
Look closely at the method signature. It has a payload (which means a HTTP body). That body contains, among other things, the filters for our query. You can't send a body with a HTTP GET (well, technically you can, but it completely screws up the HTTP caching concept of GETs because those caches don't take the body of a GET into account). So, we need to do a POST to 
be able to pass a query-filter. Same interface as this one actually:  https://github.com/kiegroup/droolsjbpm-integration/blob/master/kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/QueryDataResource.java#L187

I could change the method name into something like "runProcessInstancesQueryWithFilters" if necessary.